packages/cli/src/oclif/commands/scaffold.js

```diff
@@ -0,0 +1,213 @@
+    const preventOverwrite = !this.flags.force;
+
+    if (preventOverwrite && fileExistsSync(destPath)) {
+      const [filename, location] = splitFileFromPath(destPath);
+
+      return this.error(
+        [
+          `File ${colors.bold(filename)} already exists within ${colors.bold(
+            location
+          )}.`,
+          'You could:',
+          '  1. Choose a different filename',
+          `  2. Delete ${filename} from ${location}`,
+          `  3. Run ${colors.italic('scaffold')} with ${colors.bold(
+            '--force'
+          )} to overwrite the current ${filename}`
+        ].join('\n')
+      );
+    }
+
+    const template = await readFile(templatePath);
+    const renderTemplate = _.template(template.toString(), {
+      interpolate: /<%=([\s\S]+?)%>/g
+    });
+
+    this.startSpinner(`Writing new ${dest}.js`);
+
+    await ensureDir(path.dirname(destPath));
+    await writeFile(destPath, renderTemplate(templateContext));
+    this.stopSpinner();
+  }
+
+  async perform() {
+    const { name, type } = this.args;
+
+    if (!typeMap[type]) {
+      return this.error(
+        `Scaffold type "${type}" not found! Please see \`zaper help scaffold\`.`
+      );
+    }
+
+    try {
+      const { defaultDest, ...templateContext } = createTemplateContext(
+        type,
+        name
+      );
+      const { dest = defaultDest, entry = 'index.js' } = this.flags;
+      const entryFile = path.join(process.cwd(), entry);
+
+      this.log(`Adding ${type} scaffold to your project.\n`);
+
+      await this.writeTemplateFile(type, templateContext, dest);
+      await this.writeTemplateFile('test', templateContext, `test/${dest}`);
+
+      const entryBuf = await readFile(entryFile);
+      this.startSpinner(`Rewriting your ${entry}`);
+
+      const lines = entryBuf.toString().split('\n');
+
+      // this is very dumb and will definitely break, it inserts lines of code
+      // we should look at jscodeshift or friends to do this instead
+
+      // insert Resource = require() line at top
+      const varName = `${templateContext.CAMEL}${camelCase(type)}`;
+      const importerLine = `const ${varName} = require('./${dest}');`;
+      lines.splice(0, 0, importerLine);
+
+      // insert '[Resource.key]: Resource,' after 'resources:' line
+      const injectAfter = `${typeMap[type]}: {`;
+      const injectorLine = `[${varName}.key]: ${varName},`;
+      const linesDefIndex = lines.findIndex(line => line.endsWith(injectAfter));
+
+      if (linesDefIndex === -1) {
+        this.stopSpinner(false);
+        return this.error(
+          [
+            `\n${colors.bold(
+              `Oops, we could not reliably rewrite your ${entry}.`
+            )} Please add:`,
+            ` * \`${importerLine}\` to the top`,
+            ` * \`${injectAfter} ${injectorLine} },\` in your app definition`
+          ].join('\n')
+        );
+      }
+
+      lines.splice(linesDefIndex + 1, 0, '    ' + injectorLine);
```

Comment:
I almost think this section could be its own section so it's even easier to swap in an AST approach eventually

Reply:
i agree!